flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -540,39 +571,41 @@ private FinalApplicationStatus getYarnStatus(ApplicationStatus status) {
 	/**
 	 * Request new container if pending containers cannot satisfy pending slot requests.
 	 */
-	private void requestYarnContainerIfRequired() {
-		int requiredTaskManagers = getNumberRequiredTaskManagers();
-
-		if (requiredTaskManagers > numPendingContainerRequests) {
-			requestYarnContainer();
-		}
+	private void requestYarnContainerIfRequired(Resource containerResource) {
+		getPendingWorkerNums().entrySet().stream()
```

Comment:
I have to admit that I find `getPendingWorkerNums()` and `pendingWorkerCounter` quite confusing. The sound almost the same but the former means the requirements of the `SlotManager` and the latter the currently pending workers which have been requested by the RM.

Reply:
What's your suggestion here? How about renaming `getPendingWorkerNums` to `getSlotManagerPendingWorkerNums`?

Reply:
maybe we could rename `getPendingWorkerNums` into `getRequiredWorkers` or `getRequiredResources`.